compound_disease/compound_treats_disease/literature_models/coco_score/coco_score.py

```diff
@@ -0,0 +1,376 @@
+)
+word_vectors.head(2)
+
+
+# In[12]:
+
+
+word_dict = pd.read_table("../../word_vector_experiment/results/compound_treats_disease_word_dict.tsv", keep_default_na=False)
+reverse_word_dict = dict(zip(word_dict.index, word_dict.word))
+word_dict = dict(zip(word_dict.word, word_dict.index))
+
+
+# In[13]:
+
+
+total_training_sentences_df = (
+    all_embedded_cd_df.merge(
+        distant_supervision_marginals
+        .assign(labels=lambda x: x.pos_class_marginals > 0.5)
+        [["labels", "candidate_id"]]
+        .astype({"labels":int}),
+        on="candidate_id"
+    )
+)
+total_training_sentences_df.head(2)
+
+
+# In[14]:
+
+
+total_dev_sentences_df = (
+    all_embedded_cd_df.merge(
+        candidate_dfs['dev']
+        [["curated_ctd", "candidate_id"]],
+        on="candidate_id"
+    )
+)
+total_dev_sentences_df.head(2)
+
+
+# In[15]:
+
+
+total_test_sentences_df = (
+    all_embedded_cd_df.merge(
+        candidate_dfs['test']
+        [["curated_ctd", "candidate_id"]],
+        on="candidate_id"
+    )
+)
+total_test_sentences_df.head(2)
+
+
+# In[16]:
+
+
+def create_data_matrix(query_df, filename="sentences.txt"):
+    search_regex = rf'(\b{word_dict["~~[[2"]}\b.+\b{word_dict["2]]~~"]}\b,)'
+    search_regex += rf'|(\b{word_dict["~~[[1"]}\b.+\b{word_dict["1]]~~"]}\b,)'
+    
+    print(search_regex)
+    data = []
+    with open(filename, "w") as g:
+        for index, row in tqdm_notebook(query_df.iterrows()):
+            cand_str = ",".join(map(str, row.dropna().astype(int).values))
+            pruned_str = re.sub(search_regex, "", cand_str)
+            values = list(map(int, pruned_str.split(",")))
+            g.write(f"__label__{values[-1]}\t")
+            g.write("\t".join([reverse_word_dict[val] for val in values[:-2]]))
+            g.write("\n")
+
+
+# In[17]:
+
+
+query_df = (
+    total_training_sentences_df
+    [[col for col in total_training_sentences_df.columns if col not in ["sen_length"]]]
+)
+create_data_matrix(query_df, filename="training.txt")
+
+
+# In[18]:
+
+
+query_df = (
+    total_dev_sentences_df
+    [[col for col in total_dev_sentences_df.columns if col not in ["sen_length"]]]
+)
+create_data_matrix(query_df, filename="dev.txt")
+
+
+# In[19]:
+
+
+query_df = (
+    total_test_sentences_df
+    [[col for col in total_test_sentences_df.columns if col not in ["sen_length"]]]
+)
+create_data_matrix(query_df, filename="test.txt")
+
+
+# In[24]:
+
+
+os.system("../../../../../fastText/fasttext supervised -input training.txt -output ctd_model -lr 0.005 -epoch 50 -dim 300 -wordNgrams 2")
+os.system("../../../../../fastText/fasttext predict-prob ctd_model.bin dev.txt > dev_predictions.tsv")
+os.system("../../../../../fastText/fasttext predict-prob ctd_model.bin test.txt > test_predictions.tsv")
+
+
+# In[25]:
+
+
+precision, recall, _ = precision_recall_curve(
+    total_dev_sentences_df.curated_ctd,
+    pd.read_table('dev_predictions.tsv', header=None, sep=" ")[1]
+)
+auc(recall, precision)
```

Comment:
So these curves are how accurately you were able to predict an edge in hetionet?

Reply:
Exactly. These curves show the varying recall and precision thresholds for each cutoff score. The better the model the higher the area under the curve is.